cordova-common/src/FileUpdater.js

```diff
@@ -0,0 +1,389 @@
+                // The target file does not exist, so it should be copied from the source.
+                log("copy " + sourcePath + " " + targetPath +
+                    (!force ? " (new file)" : ""));
+                shell.cp("-f", sourceFullPath, targetFullPath);
+                updated = true;
+            }
+        } else if (sourceStats.isFile() && targetStats.isFile() &&
+                (force || sourceStats.mtime > targetStats.mtime)) {
+            // When the source and target paths both exist and are files, update
+            // the file if the source is newer or if doing a forced update.
+            log("copy " + sourcePath + " " + targetPath +
+                (!force ? " (updated file)" : ""));
+            shell.cp("-f", sourceFullPath, targetFullPath);
+            updated = true;
+        }
+    } else if (targetStats) {
+        // The target exists but the source is null, so the target should be deleted.
+        if (targetStats.isDirectory()) {
+            log("rmdir " + targetPath + " (no source)");
+            shell.rm("-rf", targetFullPath);
+        } else {
+            log("delete " + targetPath + " (no source)");
+            shell.rm("-f", targetFullPath);
+        }
+        updated = true;
+    }
+
+    return updated;
+}
+
+/**
+ * Updates a target file or directory with a source file or directory. (Directory updates are
+ * not recursive.)
+ *
+ * @param {string|null} rootDir Root directory (such as a project) to which target and source
+ *     path parameters are relative, or null if the paths are absolute. The rootDir is omitted
+ *     from any logged paths, to make the logs easier to read.
+ * @param {string} targetPath Destination file or directory to be updated. If it does not exist,
+ *     it will be created.
+ * @param {string|null} sourcePath Source file or directory to be used to update the
+ *     destination. If the source is null, then the destination is deleted if it exists.
+ * @param {boolean} force If target and source are both files, and the force flag is not
+ *     set, then the file will not be copied unless the source is newer than the target.
+ * @param {function} [log] Optional logging callback that takes a string message describing any
+ *     file operations that are performed.
+ * @return {boolean} true if any changes were made, or false if the force flag is not set
+ *     and everything was up to date
+ */
+function updatePath(rootDir, targetPath, sourcePath, force, log) {
+    rootDir = rootDir || "";
+    if (typeof(rootDir) !== "string") {
+        throw new Error("A root directory path is required.");
+    }
+
+    if (!targetPath || typeof(targetPath) !== "string") {
+        throw new Error("A target path is required.");
+    }
+
+    if (sourcePath && typeof(sourcePath) !== "string") {
+        throw new Error("A source path (or null) is required.");
+    }
+
+    log = log || function(message) { };
+
+    var targetFullPath = path.join(rootDir, targetPath);
```

Comment:
IMO `path.resolve` is more appropriate here, as it will handle correctly the case when someone has passed both `rootDir` and absolute `targetPath`. `path.join` will return incorrect path in this case.